### ISSUE TITLE
Support spaCy blank tokenizers

### ISSUE THREAD

**Issue by koaning** [RasaHQ/rasa#8108]
When you go to the spaCy documentation page on [models](https://spacy.io/usage/models) you'll see a long list of models. Some of them support a language model while other languages only support a tokeniser. 

If you want a pretrained model with vectors then you'd load it in like so; 

```python
import spacy
nlp = spacy.load("en_core_web_md")
```

If you want a spaCy model with just a tokeniser you can use the `spacy.blank` syntax.

```python
import spacy
nlp = spacy.blank("fi")
```

Some of these `blank` models feature lemmatization too! Which can be incredibly useful for languages like Turkish. At the moment however, we don't support the blank models in Rasa. We only support the pre-trained versions. By adding this feature we suddenly support 50 extra languages with tokenisation. Some of these, like Thai, would benefit a lot. 

#### Option 1 
 
We expand our `SpacyNLP` object to also support this. When a user passes a two letter language we could assume the blank model. There's a downside to this approach. This will be confusing to users who are currently using the `spacy link en en_core_web_md` feature. Note that `spacy link` is deprecated in spaCy 3.0 but it is something a lot of user relied on for spaCy 2.0. 

#### Option 2 

We create a new spaCy object: `SpacyBlankNLP`. This is the lightweight spaCy object that only supports tokenisation. 

#### Option 3 

We move this feature to rasa-NLU-examples for the time being to keep this repository more lightweight.

@dakshvar22 @Ghostvv @wochinge any opinions?

**Comment by wochinge** [comment]
Would that blank model be re-used by any other components? What speaks against supporting this in `SpacyTokenizer`?

**Comment by koaning** [comment]
@wochinge it'd involve a rewrite of the `Tokenizer`. It might also spark confusion since it breaks any spaCy component that follows. I've pinged some folks from research and we figured that a first step would be to implement this over at [rasa nlu examples](https://github.com/RasaHQ/rasa-nlu-examples). Should the demand for this tool increase (maybe because spaCy started supporting more languages) then we can consider porting it back in. 

Closing for now.
